problem-solver/cxx/format-translators-module/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+file(GLOB SOURCES CONFIGURE_DEPENDS
+	"*.cpp" "*.hpp"
+	"agent/*/*.cpp" "agent/*/*.hpp"
+	"keynodes/*.hpp"
+	"model/*.cpp" "model/*.hpp"
+	"visitor/*.cpp" "visitor/*.hpp"
+)
+
+add_library(format-translators SHARED ${SOURCES})
+target_link_libraries(format-translators LINK_PUBLIC
+	sc-machine::sc-memory
+	sc-machine::sc-core
+	sc-machine::sc-agents-common
+)
+target_include_directories(format-translators PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(SC_EXTENSIONS_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/extensions)
+set_target_properties(format-translators PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${SC_EXTENSIONS_DIRECTORY})
+
+if (${SC_BUILD_TESTS})
+	file(GLOB TEST_SOURCES CONFIGURE_DEPENDS
+		"test/units/*.cpp"
+	)
+	add_definitions(-DFORMAT_TRANSLATORS_MODULE_TEST_SRC_PATH="${CMAKE_CURRENT_SOURCE_DIR}/test")
```

Comment:
We don't use add_definitions in cmake any more. Configure tests in pre-execution stages instead of compiler stage